docs/swift/DesignDoc.md

```diff
@@ -95,13 +117,13 @@ unique type symbol emitted in the generated binary.
 
 Users are likely to balk at the ugliness of underscore-delimited names for every
 generated type. To improve upon this situation, we will add a new string file
-level option, `swift_package_typealias`, that can be added to `.proto` files.
-When present, this will cause `typealias`es to be added to the generated Swift
-messages that replace the package name prefix with the provided string. For
-example, the following `.proto` file:
+level option, `swift_package_prefix`, that can be added to `.proto` files.
+When present, this will cause the package portion of the top-level type names
+to be replaced with the provided string. For example, the following `.proto`
+file:
 
 ```protobuf
-option swift_package_typealias = "FBP";
+option swift_package_prefix = "FBP";
```

Comment:
Now that I look at this, _package_ is sorta overload between Swift and protos. Should we call this `swift_proto_package_prefix`?  The objc one is `objc_class_prefix` (and also applies to enums), but given the Swift class/struct differences, that doesn't seem like it helps too much.  `swift_prefix` might also work (if the protobuf team signs off on it).